File: 9. Palindrome Number/By Number/palindrome_number_by_number.cs
using System;

namespace palindrome_number_by_number_ns
{
	class palindrome_number_by_number_c
	{
	static bool negative_as_i_squared = true;
	
	static int reverse_number (int number){
		int base_radix = 10;
		int nh = number;
		int i = 1;
		int nje = 0;
		int mod;
		
		while (nh > 0){
			nje *= i;
			mod = nh % base_radix;
			nje += mod;
			nh -= mod;
			nh /= base_radix;
			i = 10;
		}
		
		return (nje);
	}
	
	static bool palindrome_checker (int number){
		bool nje = false;
		bool negative = (number < 0);
		
		number = (number < 0) ? (number * -1) : (number * 1);
		
		int reversed_number = reverse_number(number);
		
		if (negative_as_i_squared){
			nje = (number == reversed_number);
		} else {
			if (negative){
				nje = ((number * -1) == reversed_number);
			} else {
				nje = (number == reversed_number);
			}
		}
		
		return (nje);
	}
	
	static void Main (string [] args){
		Console.WriteLine("This is the handy work of C#");
		
		Console.WriteLine(palindrome_checker(232321));
	}
	}
}

File: 9. Palindrome Number/By Number/palindrome_number_by_number.rs
static negative_as_i_squared: bool = true;

fn reverse_number (number: isize) -> isize{
	let base: isize = 10;
	let mut nh: isize = number;
	let mut i: isize = 1;
	let mut out: isize = 0;
	let mut modulus: isize;
	
	while nh > 0{
		out *= i;
		modulus = nh % base;
		out += modulus;
		nh -= modulus;
		nh /= base;
		i = 10;
	}
	
	return out;
}

fn palindrome_checker (mut number: isize) -> bool{
	let mut out: bool = false;
	let negative: bool = number < 0;
	
	number = if number < 0 {number * -1} else {number * 1};
	
	let reversed_number: isize = reverse_number (number);
	let nais: bool;
	
	unsafe{
		nais = negative_as_i_squared;
	}
	
	if nais {
		out = number == reversed_number
	} else {
		if negative{
			out = (number * -1) == reversed_number;
		} else {
			out = number == reversed_number;
		}
	}
	
	return out;
}

fn main (){
	println!("This is the handy work of Rust");
	
	println!("{}\n\n", palindrome_checker(232321));
}
